apps/coscrad-frontend/src/store/slices/resources/shared/selectors/create-resource-selector.ts

```diff
@@ -0,0 +1,62 @@
+import { IBaseViewModel, IIndexQueryResult, ResourceType } from '@coscrad/api-interfaces';
+import { createSelector } from '@reduxjs/toolkit';
+import { ILoadable } from '../../../interfaces/loadable.interface';
+import { selectLoadableBibliographicReferences } from '../../bibliographic-references/selectors';
+import { selectLoadableBooks } from '../../books/selectors';
+import { selectLoadableMediaItems } from '../../media-items/selectors';
+import { selectLoadablePhotographs } from '../../photographs';
+import { selectLoadableSongs } from '../../songs/selectors';
+import { selectLoadableSpatialFeatures } from '../../spatial-features/selectors';
+import { selectLoadableTerms } from '../../terms/selectors';
+import { selectLoadableTranscribedAudioItems } from '../../transcribed-audio/selectors';
+import { selectLoadableVocabularyLists } from '../../vocabulary-lists/selectors';
+
+type AllResources = {
+    [K in ResourceType]: ILoadable<IIndexQueryResult<IBaseViewModel>>;
+};
+
+const emptyLoadable: ILoadable<IBaseViewModel[]> = {
+    isLoading: false,
+    errorInfo: null,
+    data: [],
+};
+
+export const createResourceSelector = () => {
```

Comment:
I don't think we used this. Thank goodness! This was really not extensible.

Reply:
Yeah, this is a mouthful

Reply:
removed